python/nano/src/bigdl/nano/pytorch/strategies/ipex/ipex_strategy.py

```diff
@@ -57,12 +57,12 @@ def setup(self, trainer: pl.Trainer) -> None:
 
         :param trainer: the trainer instance
         """
-        self.setup_precision_plugin()
```

Comment:
Could you explain a little bit why changing here?

Reply:
there are some strange things here: when using ddp_spawn or ddp_subprocess, if we call `trainer.test()`, then the `len(self.optimizers)` will be 0, so we need to handle this case.

here we don't take this case into consideration before, so I changed it to handle this case, but the strange thing is that when using IPEX strategy, the `len(self.optimizers)` is still 1 when we call `trainer.test()`.

so it seems this change is not necessary.

in addition, use `super().setup()` to replace `self.setup_precision_plugin()` and `self.setup_optimizers()` for reliability